File: container2/leetcodely/python/reverse_integer.py
"""Given a 32-bit signed integer, reverse digits of an integer."""


class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """

        if str(x)[0] == '-':
            res = -int(str(x)[1:][::-1])
        else:
            res = int(str(x)[::-1])

        if (-1 << 31) < res < (1 << 31):
            return res
        else:
            return 0


if __name__ == '__main__':
    solution = Solution()
    print(solution.reverse(-123))

File: leetcodely/javascript/src/reverseInteger.js
/*Given a 32-bit signed integer, reverse digits of an integer.*/

const reverse = x => {
    if(x === 0) {
        return 0;
    }

    let a = [...x.toString()];
    let arr;
    if(a[0] === '-'){
        arr = a.slice(1).reverse();
        arr.unshift('-')
        console.log(arr);

    }else if(a[0] === '0'){
        arr = a.slice(1).reverse();
    }else{
        arr = a.reverse()
    }
    const res = parseInt(arr.join(''));
    if(res > 2147483648 || res < -2147483648){
        return 0;
    }else{
        return res;
    }

};
console.log(reverse(-123));